roboteq_diff_driver/src/driver.cpp

```diff
@@ -218,329 +24,132 @@ MainNode::MainNode() :
   current_last_time(0),
 #endif
   pub_odom_tf(true),
-  open_loop(false),
   baud(115200),
-  wheel_circumference(0),
+  wheel_radius(0),
   track_width(0),
   encoder_ppr(0),
   encoder_cpr(0),
   max_amps(0.0),
   max_rpm(0),
-  roboteq_rpm_scale(1),
   nhLocal("~"),
   ros_loop_rate(1.0),
-  serial_timeout_ms(1.0)
+  serial_timeout_ms(1.0),
+  torque_right_meas(0.0),
+  torque_left_meas(0.0),
+  debug_(false)
 {
-
-
   // CBA Read local params (from launch file)
   nhLocal.param("pub_odom_tf", pub_odom_tf, true);
-  ROS_INFO_STREAM("pub_odom_tf: " << pub_odom_tf);
   nhLocal.param<std::string>("odom_frame", odom_frame, "odom");
-  ROS_INFO_STREAM("odom_frame: " << odom_frame);
   nhLocal.param<std::string>("base_frame", base_frame, "base_link");
-  ROS_INFO_STREAM("base_frame: " << base_frame);
-  nhLocal.param<std::string>("cmdvel_topic", cmdvel_topic, "cmd_vel");
-  ROS_INFO_STREAM("cmdvel_topic: " << cmdvel_topic);
+  nhLocal.param<std::string>("rpm_topic", rpm_topic, "cmd_rpm");
   nhLocal.param<std::string>("odom_topic", odom_topic, "odom");
-  ROS_INFO_STREAM("odom_topic: " << odom_topic);
   nhLocal.param<std::string>("port", port, "/dev/ttyACM0");
-  ROS_INFO_STREAM("port: " << port);
   nhLocal.param("baud", baud, 115200);
-  ROS_INFO_STREAM("baud: " << baud);
-  nhLocal.param("open_loop", open_loop, false);
-  ROS_INFO_STREAM("open_loop: " << open_loop);
-  nhLocal.param("wheel_circumference", wheel_circumference, 0.3192);
-  ROS_INFO_STREAM("wheel_circumference: " << wheel_circumference);
   nhLocal.param("track_width", track_width, 0.4318);
-  ROS_INFO_STREAM("track_width: " << track_width);
   nhLocal.param("encoder_ppr", encoder_ppr, 900);
-  ROS_INFO_STREAM("encoder_ppr: " << encoder_ppr);
   nhLocal.param("encoder_cpr", encoder_cpr, 3600);
-  ROS_INFO_STREAM("encoder_cpr: " << encoder_cpr);
-  nhLocal.param("max_amps", max_amps, 5.0);
-  ROS_INFO_STREAM("max_amps: " << max_amps);
+  nhLocal.param("max_amps", max_amps, 400.0);
   nhLocal.param("max_rpm", max_rpm, 100);
-  ROS_INFO_STREAM("max_rpm: " << max_rpm);
-  nhLocal.param("roboteq_rpm_scale", roboteq_rpm_scale, 10.0);
-  ROS_INFO_STREAM("roboteq_rpm_scale: " << roboteq_rpm_scale);
   nhLocal.param("ros_loop_rate", ros_loop_rate, 500.0);
-  ROS_INFO_STREAM("ros_loop_rate: " << ros_loop_rate);
   nhLocal.param("serial_timeout_ms", serial_timeout_ms, 20.0);
-  ROS_INFO_STREAM("serial_timeout_ms: " << serial_timeout_ms);
-
+  nhLocal.param("wheel_radius", wheel_radius, 0.25);
+  nhLocal.param("override_torque", override_torque, true);
+  nhLocal.param("drive_mode", drive_mode, 5);
+  nhLocal.param("debug", debug_, false);
 }
 
-
-//
-// cmd_vel subscriber
-//
-
-void MainNode::cmdvel_callback( const geometry_msgs::Twist& twist_msg)
+void MainNode::rpm_callback(const roboteq_diff_msgs::WheelRates& rpm_msg)
```

Comment:
If I understand correctly that this is called to do both RPM control and torque control, then this function needs to be renamed.

Reply:
Edited accordingly